### PR TITLE
ci: fix release dry-run CI

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

release dry-run CI after the upgrade of alpine version in dockerfiles
- in this pull request, I update the docker verification script:
  - added BASE_IMAGE placeholder in the golden file 
  - extract the base image from the dockerfile

I also updated the CI so that it can be manually run on workflow_dispatch from any other branch and not only `main`
Slack notification is skipped in case the CI is triggered with workflow_dispatch to avoid the noise in the channel

release dry-run on this branch: https://github.com/camunda/camunda/actions/runs/10075985777

## Related issues

closes #

### CODE REVIEW

File: .github/workflows/camunda-platform-release-main-dry-run.yml [hunk 1]
```diff
@@ -1,13 +1,19 @@
 name: Camunda Platform Release Dry Run from main
 on:
-  workflow_dispatch: { }
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'Specifies the branch, for which the Release Dry Run should be executed'
+        default: 'main'
+        required: false
+        type: string
```
Comment (lenaschoenburg) [inline .github/workflows/camunda-platform-release-main-dry-run.yml:9]: :+1: Nice addition for future testing

File: .ci/docker/test/verify.sh [hunk 2]
```diff
@@ -90,6 +90,16 @@ else
     exit 1
 fi
 
+BASE_IMAGE_REGEX='ARG BASE_IMAGE="([^"]+)"'
+DOCKERFILE=$(<"${BASH_SOURCE%/*}/../../../${DOCKERFILENAME}")
```
Comment (lenaschoenburg) [inline .ci/docker/test/verify.sh:94]: :upside_down_face: We could just define this once (see line 84). Would then also be a good opportunity to gracefully handle the case that this file doesn't exist.

**Review by lenaschoenburg — APPROVED** [review]
Thanks! :rocket:

File: .ci/docker/test/verify.sh (hunk 2 above)
Comment (houssain-barouni) [reply .ci/docker/test/verify.sh:94]: right, came with copy pasting 😄

**Pull request merged**